src/function/table/system/duckdb_functions.cpp

```diff
@@ -646,9 +648,136 @@ bool ExtractFunctionData(FunctionEntry &entry, idx_t function_idx, DataChunk &ou
+		parameter_names = Value::LIST(LogicalType::VARCHAR, {Value("num_buckets")});
+		break;
+	}
+	case ExpressionType::WINDOW_PERCENT_RANK:
+	case ExpressionType::WINDOW_CUME_DIST: {
+		return_type = Value("DOUBLE");
+		parameter_types = Value::LIST(LogicalType::VARCHAR, {});
+		parameter_names = Value::LIST(LogicalType::VARCHAR, {});
+		break;
+	}
+	case ExpressionType::WINDOW_LEAD:
+	case ExpressionType::WINDOW_LAG: {
+		return_type = Value("T");
+		parameter_types = Value::LIST({Value("T"), Value("BIGINT"), Value("T")});
+		parameter_names = Value::LIST({Value("expr["), Value("offset["), Value("default]]")});
+		break;
+	default:
+		return;
+	}
+	}
+
+	idx_t col = 0;
+
+	// database_name, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value(SYSTEM_CATALOG));
+
+	// database_oid, BIGINT
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::BIGINT));
+
+	// schema_name, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value(DEFAULT_SCHEMA));
+
+	// function_name, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, name);
+
+	// alias_of, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::VARCHAR));
+
+	// function_type, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value("window"));
+
+	// function_description, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::VARCHAR));
+
+	// comment, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::VARCHAR));
+
+	// tags, LogicalType::MAP(LogicalType::VARCHAR, LogicalType::VARCHAR)
+	output.SetValue(col++, output_offset, Value::MAP({}));
+
+	// return_type, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, return_type);
+
+	// parameters, LogicalType::LIST(LogicalType::VARCHAR)
+	output.SetValue(col++, output_offset, parameter_names);
+
+	// parameter_types, LogicalType::LIST(LogicalType::VARCHAR)
+	output.SetValue(col++, output_offset, parameter_types);
+
+	// varargs, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::VARCHAR));
+
+	// macro_definition, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::VARCHAR));
+
+	// has_side_effects, LogicalType::BOOLEAN
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::VARCHAR));
+
+	// internal, LogicalType::BOOLEAN
+	output.SetValue(col++, output_offset, Value::BOOLEAN(true));
+
+	// function_oid, LogicalType::BIGINT
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::BIGINT));
+
+	// examples, LogicalType::LIST(LogicalType::VARCHAR)
+	output.SetValue(col++, output_offset, Value::LIST(LogicalType::VARCHAR, {}));
+
+	// stability, LogicalType::VARCHAR
+	output.SetValue(col++, output_offset, Value(LogicalTypeId::VARCHAR));
+
+	// categories, LogicalType::LIST(LogicalType::VARCHAR)
+	output.SetValue(col++, output_offset, Value::LIST(LogicalType::VARCHAR, {}));
```

Comment:
Is it possible to not duplicate the code that sets these columns by implementing a custom extractor like we did for scalar/aggregate/macro/tablemacro/tablefunction?

Reply:
I don't want to complicate the `ExtractFunctionData` method with further templating that should be later stripped down again

*maybe* I can create some dummy data in a shape that ExtractFunctionData accepts, I can try that

Reply:
Ah, I see. Only half the stuff is templated using an extractor, the rest is taken from the `FunctionEntry`, which does not yet exist for Window. Then it makes sense to duplicate this until we implement that